compiler/resolution/resolveFunction.cpp

```diff
@@ -1382,10 +1382,20 @@ static void markTempsDeadLastMention(std::set<VarSymbol*>& temps) {
         // returning into a user var?
         if (lhsSe != NULL) {
           VarSymbol* lhs = toVarSymbol(lhsSe->symbol());
-          if (lhs != NULL && lhs != v && !lhs->hasFlag(FLAG_TEMP)) {
+          if (lhs != NULL && lhs != v) {
+
             // Used in initializing a user var, so mark end of block
-            makeThemEndOfBlock = true;
-            break;
+            if (!lhs->hasFlag(FLAG_TEMP)) {
+              makeThemEndOfBlock = true;
+              break;
+
+            // For e.g. 'manage new foo() do ...;'
+            // The manager handle is a temp that refers to 'new foo()', so
+            // what it refers to should live until the end of the block.
```

Comment:
How about we make it not a temp from the beginning? Would it allow us to avoid this special casing?

Reply:
I could not make it a temp, but then I would have to make a variable prefixed with `chpl_`? How else to avoid infringing on the user namespace?

Reply:
I am not sure if I follow.

You can prefix it with `chpl_`, but whether you do it or not, you cannot mess the name space by picking wrong names for compiler-injected variables. You just need to create a symbol and put it in a `SymExpr` and use that expr. I think what you're worried is that `scopeResolve` gets messed up, but as long as the SymExpr is referring to the symbol you create, you should be fine.

And still, none of this matters unless it helps remove this special case. If it has to stay for some other reason, I'm ok with that variable being a temp.